### PR TITLE
Implement admin user management

### PR DESCRIPTION
**Pull request by allenwq**

### CODE REVIEW

File: app/controllers/system/admin/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class System::Admin::UsersController < System::Admin::Controller
+  load_and_authorize_resource :user, class: User.name, parent: false
```
Comment (lowjoel) [inline app/controllers/system/admin/users_controller.rb]: is `parent: false` necessary?

File: app/models/user.rb [hunk 2]
```diff
@@ -22,6 +22,8 @@ class User < ActiveRecord::Base
 
   accepts_nested_attributes_for :emails
 
+  scope :sorted_by_name, -> { order(:name) }
```
Comment (lowjoel) [inline app/models/user.rb]: `ordered_by_name`?

File: app/models/user.rb [hunk 3]
```diff
@@ -66,7 +68,7 @@ def unset_primary_email
   def set_next_email_as_primary
     return false unless default_email_record
 
-    default_email_record.update_attributes(primary: true)
+    default_email_record.destroyed? || default_email_record.update_attributes(primary: true)
```
Comment (lowjoel) [inline app/models/user.rb]: maybe we should change the behaviour of default_email_record never to return a destroyed record?

Comment (allenwq) [inline app/models/user.rb]: Yep, I was about to discuss this with you.  Will do this in another PR, thanks !

**Comment by allenwq** [comment]
Depends on #439

**Comment by allenwq** [comment]
@lowjoel Can this be merged ?  Don't know why coverage changed, all files related to this PR  are covered.
